rts-front/src/screens/register.rs

```diff
@@ -0,0 +1,144 @@
+use reqwasm::http::{Request, Response};
+use serde::{Deserialize, Serialize};
+use web_sys::HtmlInputElement;
+use yew::prelude::*;
+use yew_router::prelude::*;
+
+use crate::routes::Route;
+use crate::utils::alert_message;
+
+#[derive(Serialize)]
+struct RegistrationRequest {
+    email: String,
+    username: String,
+    password: String,
+}
+
+#[derive(Deserialize)]
+#[serde(untagged)]
+enum RegistrationResponse {
+    Successful,
+    Failed(String),
+}
+
+enum ResultOrResponse {
+    Result(RegistrationResponse),
+    Response(Response),
+}
+
+#[function_component(Registration)]
+pub fn registration_screen() -> Html {
+    let email = use_state(|| "".to_string());
+    let username = use_state(|| "".to_string());
+    let password = use_state(|| "".to_string());
+    let password_confirm = use_state(|| "".to_string());
+    let history = use_history().expect("???");
+
+    let on_email_change = {
+        let email = email.clone();
+        Callback::from(move |e: Event| {
+            let input: HtmlInputElement = e.target_unchecked_into();
```

Comment:
Should we consider using the checked version ?

Reply:
https://yew.rs/docs/concepts/html/events#using-jscast

From what I understand, we don't need to use the checked version since the callbacks are from the element directly and not children elements.

The checked version would only add overhead during runtime.